hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/analysis/HoodieAnalysis.scala

```diff
@@ -413,10 +416,13 @@ case class HoodieResolveReferences(sparkSession: SparkSession) extends Rule[Logi
     // Resolve Delete Table
     case DeleteFromTable(table, condition)
       if sparkAdapter.isHoodieTable(table, sparkSession) && table.resolved =>
-      // Resolve condition
-      val resolvedCondition = condition.map(resolveExpressionFrom(table)(_))
+      val unwrappedCondition: Expression = condition match {
```

Comment:
[SPARK-38626](https://github.com/apache/spark/pull/35941) Make condition in DeleteFromTable plan required

Reply:
Let's hide this compatibility juggling w/in the SparkAdapter actually: we can create method like 

```
def resolveDeleteFromCommand(dfc: DeleteFromCommand)
```
And handle compatibility in there